File: lucafontanili.designpatterns.java/src/main/java/com/lucafontanili/designpatterns/behavioral/strategy/StrategyClient.java
package com.lucafontanili.designpatterns.behavioral.strategy;

import java.io.IOException;

import com.lucafontanili.designpatterns.utilities.AbstractClient;
import com.lucafontanili.designpatterns.utilities.InvalidArgumentException;

public class StrategyClient extends AbstractClient {

    public static void main(String[] args) throws InvalidArgumentException, IOException {
	main(new StrategyClient());
    }

    @Override
    public void run() throws InvalidArgumentException, IOException {
	ShoppingCart shoppingCart = new ShoppingCart();
	shoppingCart.addItem("Pasta", 1.0);
	shoppingCart.addItem("Basil", 0.5);
	shoppingCart.addItem("Tomatoes", 0.7);

	shoppingCart.pay(PaymentMethod.CREDIT_CARD, "xxxx-xxxx-xxxx-xxxx");
	shoppingCart.pay(PaymentMethod.PAYPAL, "test@paypal.com");
    }

}

File: lucafontanili.designpatterns.python/src/behavioral/strategy/StrategyClient.py
'''
Created on Nov 27, 2016

@author: Admin
'''

from Strategy import *
from src.behavioral.strategy import Strategy

def main():
    cart = ShoppingCart()
    cart.add_item('Pasta', 1.0)
    cart.add_item('Basil', 0.5)
    cart.add_item('Tomatoes', 0.7)
    
    cart.pay(Strategy.payment_methods['credit_card'], 'xxxx-xxxx-xxxx-xxxx')
    cart.pay(Strategy.payment_methods['paypal'], 'test@paypal.com')

if __name__ == "__main__":
    main()